subscription/protocol.go

```diff
@@ -0,0 +1,112 @@
+// Package subscription implements the Subscription NEX protocol
+package subscription
+
+import (
+	"fmt"
+
+	nex "github.com/PretendoNetwork/nex-go"
+	"github.com/PretendoNetwork/nex-protocols-go/globals"
+)
+
+const (
+	// SubscriptionProtocolID is the protocol ID for the Subscription protocol
+	SubscriptionProtocolID = 0x75
+
+	// SubscriptionCreateMySubscriptionDataID is the method ID for the method SubscriptionCreateMySubscriptionData
+	SubscriptionCreateMySubscriptionDataID = 0x1
+
+	// SubscriptionUpdateMySubscriptionDataID is the method ID for the method SubscriptionUpdateMySubscriptionData
+	SubscriptionUpdateMySubscriptionDataID = 0x2
+
+	// SubscriptionClearMySubscriptionDataID is the method ID for the method SubscriptionClearMySubscriptionDataID
+	SubscriptionClearMySubscriptionDataID = 0x3
+
+	// SubscriptionAddTargetID is the method ID for the method SubscriptionAddTarget
+	SubscriptionAddTargetID = 0x4
+
+	// SubscriptionDeleteTargetID is the method ID for the method SubscriptionDeleteTarget
+	SubscriptionDeleteTargetID = 0x5
+
+	// SubscriptionClearTargetID is the method ID for the method SubscriptionClearTarget
+	SubscriptionClearTargetID = 0x6
+
+	// SubscriptionGetFriendSubscriptionDataID is the method ID for the method SubscriptionGetFriendSubscriptionData
+	SubscriptionGetFriendSubscriptionDataID = 0x7
+
+	// SubscriptionGetTargetSubscriptionDataID is the method ID for the method SubscriptionGetTargetSubscriptionData
+	SubscriptionGetTargetSubscriptionDataID = 0x8
+
+	// SubscriptionGetActivePlayerSubscriptionDataID is the method ID for the method SubscriptionGetActivePlayerSubscriptionData
+	SubscriptionGetActivePlayerSubscriptionDataID = 0x9
+
+	// SubscriptionGetSubscriptionDataID is the method ID for the method SubscriptionGetSubscriptionData
+	SubscriptionGetSubscriptionDataID = 0xA
+
+	// SubscriptionReplaceTargetAndGetSubscriptionDataID is the method ID for the method SubscriptionReplaceTargetAndGetSubscriptionData
+	SubscriptionReplaceTargetAndGetSubscriptionDataID = 0xB
+
+	// SubscriptionSetPrivacyLevelID is the method ID for the method SubscriptionSetPrivacyLevel
+	SubscriptionSetPrivacyLevelID = 0xC
+
+	// SubscriptionGetPrivacyLevelID is the method ID for the method SubscriptionGetPrivacyLevel
+	SubscriptionGetPrivacyLevelID = 0xD
+
+	// SubscriptionGetSubscriptionUserFriendListID is the method ID for the method SubscriptionGetSubscriptionUserFriendList
+	SubscriptionGetSubscriptionUserFriendListID = 0xE
+
+	// SubscriptionGetPrivacyLevelsID is the method ID for the method SubscriptionGetPrivacyLevels
+	SubscriptionGetPrivacyLevelsID = 0xF
+)
```

Comment:
I didn't catch these at first, but these prefixes are wrong. For the protocol IDs we just leave it as `ProtocolID`, and for the method IDs we prefix them with `Method` not the name of the protocol